### ISSUE TITLE
Spectral does not distinquish paths with different path variables

### ISSUE THREAD

**Issue by adamzareba** [stoplightio/spectral#1945]
**Chore summary**
Spectral linter throws an error in case of OpenApi 3.0 specification contains the same paths with different path variable name. 
```
 32:19  error  path-params  The paths `/pets/{petId}` and `/pets/{ownerId}` are equivalent.  paths./pets/{ownerId}
```

**Tasks**
- define OpenAPI specification, like:
```
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
  description: Swagger Petstore
  contact:
    name: Swagger Petstore

tags:
  - name: pets
    description: Pets

paths:
  /pets/{petId}:
    get:
      operationId: showPetById
      description: Info for a specific pet
      tags:
        - pets
      parameters:
        - name: petId
          in: path
          required: true
          description: The id of the pet to retrieve
          schema:
            type: string
      responses:
        '200':
          description: Expected response to a valid request
        default:
          description: unexpected error
  /pets/{ownerId}:
    put:
      operationId: updateOwnersPets
      description: Updates pets
      parameters:
        - name: ownerId
          in: path
          required: true
          description: The id of the pet's owner
          schema:
            type: string
      tags:
        - pets
      responses:
        '201':
          description: Null response
        default:
          description: unexpected error
```
- Run validation: 
```
spectral lint petstore.yaml
```

**Additional context**
Spectral (5.9.1)
OS: Windows

**Comment by fednot-icc** [comment]
This behaviour is conform the swagger standard:
https://swagger.io/docs/specification/2-0/paths-and-operations/#:~:text=Swagger%20defines%20a%20unique%20operation,have%20no%20effect%20on%20uniqueness).

And OpenAPI 3:
https://swagger.io/docs/specification/paths-and-operations/
